src/bindgen/library.rs

```diff
@@ -266,6 +268,55 @@ impl Library {
         }
     }
 
+    fn set_ctype(&mut self) {
+        if self.config.style.generate_typedef() {
+            return;
+        }
+
+        let mut struct_names = HashSet::new();
```

Comment:
These should probably be made into some kind of cache that can be sent  around with convenience methods.